app/controllers/green_lanes/results_controller.rb

```diff
@@ -2,66 +2,67 @@ module GreenLanes
   class ResultsController < ApplicationController
     include GreenLanesHelper
 
-    before_action :check_green_lanes_enabled,
-                  :disable_switch_service_banner,
-                  :disable_search_form
+    before_action :set_up, only: :show
 
     def show
-      goods_nomenclature = GreenLanes::GoodsNomenclature.find(
-        results_params[:commodity_code],
-        { filter: {
-          geographical_area_id: results_params[:country_of_origin],
-          moving_date: results_params[:moving_date],
-        } },
-        { authorization: TradeTariffFrontend.green_lanes_api_token },
-      )
+      @next_page = determine_next_page
+    end
+
+    private
+
+    def set_up
+      check_green_lanes_enabled
+      disable_switch_service_banner
+      disable_search_form
+      goods_nomenclature
+      set_instance_variables
+      determine_assessments
+    end
+
+    def results_params
+      params.permit(:commodity_code, :country_of_origin, :moving_date, :c1ex, :c2ex, :category, :ans)
+    end
+
+    def goods_nomenclature
+      @goods_nomenclature ||= GreenLanes::FetchGoodsNomenclature.new(results_params).call
+    end
 
-      @commodity_code = goods_nomenclature.goods_nomenclature_item_id
+    def set_instance_variables
```

Comment:
not much in favour of this pattern.
`set_instance_variables` is called in just one place.

Reply:
happy to discuss